crowbar_framework/app/models/service_object.rb

```diff
@@ -142,12 +142,18 @@ def random_password(size = 12)
 #
 # Locking Routines
 #
+  def new_lock(name)
+    Crowbar::Lock::LocalBlocking.new(name: name, logger: @logger)
+  end
+
   def acquire_lock(name)
-    FileLock.acquire(name, logger: @logger)
+    new_lock(name).acquire
   end
 
-  def release_lock(f)
-    FileLock.release(f, logger: @logger)
+  def with_lock(name)
+    new_lock(name).with_lock do
+      yield
+    end
   end
```

Comment:
We have these methods here and in deployment_queue.rb. Should they somehow be shared?